### DISCUSSION TITLE
Signing tags created via releases

### DISCUSSION THREAD

**Discussion by AshleighAdams** [community/community#5508] [Other Feature Feedback, Questions, & Ideas] [answered]
Some people, me included, prefer to operate only on signed commits and tags. Creating a tag on GitHub.com results in an unsigned non-annotated tag.

It would be nice if these tags were signed in the same way as commits created on GitHub are, or have the option of signing it and possibly annotating it with the release title and body.

**Comment by MylesBorins** [comment, marked as answer]
@AshleighAdams thanks for the suggestion! I've been working on scoping this exact feature and we have an engineer looking into it right now 🎉

**Reply by kbond** [reply]
Probably already on your radar but I just wanted to note that tags created via the releases API are also currently unsigned. It would be nice if these could be signed too.

**Reply by MylesBorins** [reply]
Thanks for pointing that out @kbond. Implementing this feature is going to be a bit more complex than I was hoping it would be so it would take a bit of time. The good news is that the way we are thinking this could be done would work for both UI + API 🎉

**Reply by mpldr** [reply]
Since you can't sign lightweight tags, are you also addressing https://github.com/github/feedback/discussions/4924 simultaneously?

**Comment by msharov** [comment]
I'd also like the reverse problem addressed: when I sign a release tag, I have to go to the GitHub page and convert it to a release before it is recognized as one. I also have to copy the release notes to it because GitHub inconveniently treats the conversion as making a new tag and deletes them. Perhaps signed tags could automatically be recognized as releases. Maybe by checking if the tag message contains something like "Release v1.1" followed by a list of changes.

**Reply by mpldr** [reply]
I don't know about everybody else, but I only tag releases and release-candidates. What else would I tag, for which I would also push the tag?

**Reply by spenserblack** [reply]
If your release tags always have the same format, then you can use a GitHub action to convert it into a release.
I personally do
```yaml
on:
  push:
    tags: [ 'v*.*.*' ]

jobs:
  release:
    steps:
      # your steps for creating a release
```
